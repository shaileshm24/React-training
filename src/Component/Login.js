
import React, { Component } from "react";
import jwt from 'jsonwebtoken';
import webdesign from '../web-design.png'

import './login.css';

export default class Login extends Component {
  constructor(props){
      super(props)
      this.state={
          email:"",
          password:""
      }
  }

  handleEmailChange = (e) => {
    if(e.target.value === "admin"){
      this.setState({email: e.target.value})
    }
      
  }
  handlePasswordChange = (e) => {
    if(e.target.value === "admin"){
    this.setState({password: e.target.value})
    }
    else{

    }
  }

  handleSubmit = () => {
    if(this.state.email === "admin" && this.state.password === "admin"){
      const authJWT = jwt.sign({email:this.state.email, pass:this.state.password},'privatekeeeyy');
      console.log(authJWT);
      window.localStorage.setItem("token",authJWT);
      window.location.replace("http://localhost:3000/")
    }
    else{
      document.getElementById("error").innerHTML="<font color='red'>Invalid username or password </font>"
    }
  }

  render() {
    return (
      <div style={{background:"#d3d3d326", height:"100vh"}}>
        <div className="container">
        <div >
        <img className="backimg" src={webdesign} height="400px" width="400px"/>
        </div>
          <div className="login-panel">
            <div className="login-header">
             <div className="input">
             <h1>Login</h1>
             <input type="text" className="inputText" placeholder="Enter Username" onChange={this.handleEmailChange.bind(this)}></input>
             <input type="password" className="inputText" placeholder="Enter Password" onChange={this.handlePasswordChange.bind(this)}></input>
             <p id="error"></p>

             </div>
             <div className="button">
             <button className="btn" onClick={this.handleSubmit}>Login</button>
             </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
