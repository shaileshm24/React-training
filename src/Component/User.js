import React, { Component} from 'react';
import { Card, Modal, Input, Button } from 'antd';
import { Icon, Form } from '@ant-design/compatible';
import './User.css'; 
import 'antd/dist/antd.css';
import '@ant-design/compatible/assets/index.css';


class User extends Component {
  constructor(props){
    super(props)
    this.state = {
      like: false,
      modal: false,
    };
  }
 

  likeButton = () => {
    this.setState({like: !this.state.like});
  };

  handleSubmit = () => {
    this.props.form.validateFieldsAndScroll((err, data) => {
      if (!err) {
        this.props.updateUserData(this.props.user.id, data);
        this.toggleModal();
      }
    });
  };

  

  toggleModal = () => {
    this.setState({ modal: !this.state.modal});
  };

  render() {
    const { user, deleteUser } = this.props;
    const { getFieldDecorator } = this.props.form;
    const { like, modal } = this.state;

    return (
      <div>
     
        <Modal title="Edit Details" visible={modal} onOk={this.handleSubmit} onCancel={this.toggleModal}>
          <Form {...{labelCol: {
            xs: { span: 24 },
            sm: { span: 8 },
          },
          wrapperCol: {
            xs: { span: 24 },
            sm: { span: 16 },
          }}}>
            <Form.Item label="Name">
              {getFieldDecorator('name', {
                initialValue: user.name,
                rules: [
                  {
                    required: true,
                    message: 'This field is required',
                  },
                ],
              })(<Input />)}
            </Form.Item>
            <Form.Item label="Email">
              {getFieldDecorator('email', {
                initialValue: user.email,
                rules: [
                  {
                    required: true,
                    message: 'This field is required',
                  },
                  {
                    type: 'email',
                    message: 'Invalid email',
                  },
                ],
              })(<Input />)}
            </Form.Item>
            <Form.Item label="Phone">
              {getFieldDecorator('phone', {
                initialValue: user.phone,
                rules: [
                  {
                    required: true,
                    message: 'This field is required',
                  },
                ],
              })(<Input />)}
            </Form.Item>
            <Form.Item label="Website">
              {getFieldDecorator('website', {
                initialValue: user.website,
                rules: [
                  {
                    required: true,
                    message: 'This field is required',
                  },
                ],
              })(<Input />)}
            </Form.Item>
          </Form>
        </Modal>
       
        <Card
          style={{ margin: 15 }}
          cover={
            <div className="cardHeadImage" style={{display: "flex"}}>
              <img
                src={`https://avatars.dicebear.com/v2/avataaars/${user.username}.svg?options[mood][]=happy`}
                alt="Avatar"
                style={{ width: 200, height: 200 }}
              />
            </div>
          }
          actions={[
            <button
              style={{ background: 'none', border: 'none', cursor: 'pointer', outline: 'none' }}
              onClick={this.likeButton}
            >
              <Icon type="heart" style={{ color: '#FF0000', fontSize: 20 }} theme={like ? 'filled' : 'outlined'} />
            </button>,
            <button
              style={{ background: 'none', border: 'none', cursor: 'pointer', outline: 'none' }}
              onClick={this.toggleModal}
            >
              <Icon type="edit" style={{ fontSize: 18 }} />
            </button>,
            <button
              style={{ background: 'none', border: 'none', cursor: 'pointer', outline: 'none' }}
              onClick={() => {
                deleteUser(user.id);
              }}
            >
              <Icon type="delete" theme="filled" style={{ fontSize: 18 }} />
            </button>,
          ]}
        >
          <h3>{user.name}</h3>
          <div style={{ display: 'flex', flexDirection: 'row' }}>
            <Icon type="mail" style={{ fontSize: '18px' }} />
            <p style={{ marginLeft: 10 }}>{user.email}</p>
          </div>
          <div style={{ display: 'flex', flexDirection: 'row' }}>
            <Icon type="phone" style={{ fontSize: '18px' }} />
            <p style={{ marginLeft: 10 }}>{user.phone}</p>
          </div>
          <div style={{ display: 'flex', flexDirection: 'row' }}>
            <Icon type="global" style={{ fontSize: '18px' }} />
            <p style={{ marginLeft: 10 }}>http://{user.website}</p>
          </div>
        </Card>
        </div>
    );
  }
}

export default Form.create()(User);
