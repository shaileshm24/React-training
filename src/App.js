import React, { Component } from "react";
import { Route, BrowserRouter as Router, Redirect } from "react-router-dom";
import { Row, Col, Button } from "antd";
import User from "./Component/User";
import axios from "axios";
import "antd/dist/antd.css";
import "./App.css";
import Login from "./Component/Login";
import UserInfo from "./UserInfo";

class App extends Component {
  render() {
    return (
      <React.Fragment>
        <Router>
          <div>
          <Route path="/login" component={Login}></Route>
         
            {window.localStorage.getItem("token") !== "" ? (
              <Route path="/" component={UserInfo}></Route>
            
            ) : (
              <Redirect to="/login"></Redirect>
            )}{" "}
          </div>
        </Router>
      </React.Fragment>
    );
  }
}

export default App;
