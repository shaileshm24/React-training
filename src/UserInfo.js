import React, { Component } from "react";
import { Route, BrowserRouter as Router } from "react-router-dom";
import { Row, Col, Button } from "antd";
import User from "./Component/User";
import axios from "axios";
import "antd/dist/antd.css";
import "./App.css";


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userData: [],
    };
  }

  async componentDidMount() {
    const response = await axios.get(
      "https://jsonplaceholder.typicode.com/users"
    );
    this.setState({ userData: response.data });
    console.log(window.localStorage);
  }

  deleteUser = (id) => {
    this.setState((prevState) => ({
      userData: prevState.userData.filter((ele) => ele.id !== id),
    }));
  };

  updateUserData = (id, data) => {
    this.setState((prevState) => ({
      userData: prevState.userData.map((ele) => {
        if (ele.id === id) return { ...ele, ...data };
        return ele;
      }),
    }));
  };

  logout = () => {
    window.localStorage.setItem("token", "");
    window.location.replace("/login");
  };

  render() {
    const { userData } = this.state;

    if (userData.length === 0) {
      return (
        <div>
          <div className="spinner">
            <div className="bounce1" />
            <div className="bounce2" />
            <div className="bounce3" />
          </div>
        </div>
      );
    }

    return (
      <React.Fragment>
      
          
            <div>
              <Button className="logout" onClick={this.logout}>
                Logout
              </Button>
              <Row>
                {userData.map((user) => (
                  <Col xs={24} sm={24} md={8} lg={8} xl={6} key={user.username}>
                    <User
                      user={user}
                      deleteUser={this.deleteUser}
                      updateUserData={this.updateUserData}
                    />
                  </Col>
                ))}
              </Row>
            </div>
      </React.Fragment>
    );
  }
}

export default App;
