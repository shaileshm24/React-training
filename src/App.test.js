import React from 'react';
import App from './App';
import Login from "./Component/Login";
import User from "./Component/User";
import UserInfo from './UserInfo';
import { shallow } from 'enzyme';

describe("rendering components", () =>{
  it("should render Login component without crashing", () => {
    shallow(<Login />);
  });
  it("should render App component without crashing", () => {
    shallow(<App />);
  });
  it("should render UserInfo component without crashing", () => {
    shallow(<UserInfo />);
  });
  it("should render User component without crashing", () => {
    shallow(<User />);
  });
});

describe("check data which is rendering", () =>{
   it("should render component in debug mode", () =>{
     const component = shallow(<UserInfo debug/>);
     expect(component).toMatchSnapshot();
   })
})