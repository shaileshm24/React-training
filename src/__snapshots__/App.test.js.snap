// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check data which is rendering should render component in debug mode 1`] = `ShallowWrapper {}`;
